src/aap_eda/services/activation/engine/kubernetes.py

```diff
@@ -471,31 +471,10 @@ def _wait_for_pod_to_start(self, log_handler: LogHandler) -> None:
                         if reason in POD_FAILED_REASONS:
                             raise ContainerImagePullError(message)
 
-                if pod_phase == "Failed" or pod_phase == "Succeeded":
-                    statuses = event["object"].status.container_statuses
-                    if (
-                        statuses
-                        and statuses[0]
-                        and statuses[0].state.terminated
-                    ):
-                        exit_code = statuses[0].state.terminated.exit_code
-                        reason = statuses[0].state.terminated.reason
-                        if exit_code != 0:
-                            error_msg = (
-                                f"Pod {pod_name} failed with "
-                                f"exit code {exit_code}"
-                            )
-                            LOGGER.error(error_msg)
-                            log_handler.write(error_msg, True)
-                            raise ContainerStartError(error_msg)
+                if pod_phase in ["Failed", "Succeeded", "Running"]:
```

Comment:
If we `break` without any output messages here when the pod fails, the manager cannot know the failure details, right?

Reply:
What's the output after you remove the handling of `exit_code != 0`?

Reply:
This the start operation. The manager only need to know if the pod has been started or not. That is the only kind of failure needed to catch, to set the activation as started or error. If there is an exit code it means the pod has been executed. The exit code or error message is handled later by the monitor task who calls get_status. Start method does not report the status of the pod, it is a binary matter: has been started or not. It has a pod_id or the pod has not been started (an exception)